src/components/CreatePlaceFlow/components/usePlaceSearch.js

```diff
@@ -0,0 +1,49 @@
+// @flow
+import * as React from 'react';
+import { searchPlacesDebounced } from '../../../lib/searchPlaces';
+import type { SearchResultCollection } from '../../../lib/searchPlaces';
+
+export type SearchState = 'TypeMore' | 'Loading' | 'Error' | 'Results';
+
+function usePlaceSearch(minimalRequiredLetterCount: number = 2) {
+  const [searchString, setSearchString] = React.useState<string>('');
+  const [searchState, setSearchState] = React.useState<SearchState>('TypeMore');
+  const [searchResults, setSearchResults] = React.useState<SearchResultCollection | null>(null);
+  const request = React.useRef<Promise<SearchResultCollection> | null>(null);
+
+  React.useEffect(() => {
+    if (searchString.length >= minimalRequiredLetterCount) {
+      setSearchState('Loading');
+      // Maybe pass current geo coords here - do we have a useUserCoords already?
```

Comment:
The problem with photon's lat/lon query would be that it gives places somewhere else no priority at all, so you don't find that stuff anymore. We stumbled upon this when implementing general search. We'd have to mix/sort results from 2 queries (one nearby, one by name) to overcome this problem, which I find too complex - so I'd suggest we use the results that work for both the nearby case and the somewhere-else case, which are the ones without geo coordinates. They're not optimal as they show totally irrelevant stuff too (it's possible but highly unlikely that somebody tags something outside their country), but that's better than not finding something that you search at all.

Reply:
Displaying the distance to a search result might also help here.